### ISSUE TITLE
Using standardized-audio-context as a global polyfill?

### ISSUE THREAD

**Issue by JohnWeisz** [chrisguttandin/standardized-audio-context#979]
Hi, and thanks for this great project.

Is it possible to use standardized-audio-context as a global polyfill somehow, without explicitly importing the wrappers?

I did try overwriting AudioContext-related keys on `window`, but it's resulting in a SyntaxError from the AudioWorklet.addModule method here: https://github.com/chrisguttandin/standardized-audio-context/blob/c77d611aabdda0a664233dbc666ecc7e7eb090be/src/factories/add-audio-worklet-module.ts#L146 _(the block catches a DOMException without an additional type or message set)_

Note: the audio worklet module itself is fine, it loads both with the native audio ctx, and also when the wrapper is used correctly through an import. It's this most probably incorrect global polyfill attempt I'm doing that causes the issue.

Thanks!

**Comment by chrisguttandin** [comment]
Hi @JohnWeisz, thanks for your nice words.

I actually never tried this but I guess it will make standardized-audio-context trip over its own feed when assigning its exports to the global window object. In some occasions the window object is used from inside the library which may cause weird behavior.

Do you have some example code of what you tried? It would make it a bit easier for me to find the actual issue.

**Comment by JohnWeisz** [comment]
I simply reassigned the exported objects, the following would be the shortest approach:

```
import * as ctx from "standardized-audio-context";

for (let key of Object.keys(ctx)) {
    (window as any)[key] = (ctx as any)[key];
}
```

Looking into the code briefly it indeed seems that standardized-audio-context expects the native Web Audio API objects to remain unaltered.

To give some context, the reason I'm asking if a global polyfill is possible is because I'm looking to try this awesome ponyfill in one of my already established projects, which would require serious changes to replace all the global references (especially so because it's TS as well). However, standardized-audio-context just happens to be the most comprehensive ponyfill.

Thanks.

**Comment by chrisguttandin** [comment]
I think 88784bd445edea988ab19e74228346897e4f3b52 solved the issue. I published this as v25.2.8.

I made a quick test using [standardized-audio-context-demo](https://github.com/chrisguttandin/standardized-audio-context-demo). It works as expected now when patching the window object as done in your snippet. It didn't work before.

Anyway standardized-audio-context-demo is only using a subset of the exports. Please let me know if you're still experiencing any issues.
